### PR TITLE
Integrate 2.0.0 SDK into RTS Viewer tvOS

### PR DESCRIPTION
**Pull request by aravind-raveendran**
##Description
1. Code refactor and reorganisation (Removed RTSComponentKit)
2. Using the package RTSCore with common 2.0 SDK consumption code 
3. Add more stream statistics
4. Display all available layers for selection named after their resolution or encoding id (layers are not more named as high, medium, low anymore)

### CODE REVIEW

**Review by Yousif-CS — APPROVED** [review]
LGTM

File: rts-viewer-tvos/RTSViewer/ReusableViews/SettingsView/SettingsView.swift [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+        .transition(.move(edge: .trailing))
+        .overlay {
+            if showSimulcastView {
+                SimulcastView(
+                    source: viewModel.source,
+                    videoQualityList: viewModel.videoQualityList,
+                    selectedVideoQuality: viewModel.selectedVideoQuality) { videoQuality in
+                        Task {
+                            showSimulcastView = false
+                            onSelectVideoQuality(viewModel.source, videoQuality)
```
Comment (Agustin-Bologna-Dolby) [inline rts-viewer-tvos/RTSViewer/ReusableViews/SettingsView/SettingsView.swift:73]:
```suggestion
                            showSimulcastView = false
                            onSelectVideoQuality(viewModel.source, videoQuality)
                            focus = .simulcastSelection
```

File: rts-viewer-tvos/RTSViewer/ReusableViews/SettingsView/SettingsView.swift [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+                        }
+                    }
+                    .onExitCommand {
+                        if showSimulcastView {
+                            showSimulcastView = false
```
Comment (Agustin-Bologna-Dolby) [inline rts-viewer-tvos/RTSViewer/ReusableViews/SettingsView/SettingsView.swift:78]:
```suggestion
                            showSimulcastView = false
                            focus = .simulcastSelection
```

**Review by Agustin-Bologna-Dolby — CHANGES_REQUESTED** [review]
While trying the app I found the following issues: 

- The streams don't have sound when they should have.
- Whenever I change the simulcast option I get redirected back to `auto`
- If the statistics are displayed on screen and you open the settings view the statistics panel does a weird effect.

File: LocalPackages/RTSCore/Sources/RTSCore/Models/SourceBuilder.swift [hunk 3]
```diff
@@ -147,63 +126,51 @@ final class SourceBuilder {
...
+                    case .inactive:
+                        Self.logger.debug("👨‍🔧 Audio track for \(sourceId) is inactive, \(track.isActive)")
+                        self.audioTrackStateUpdateSubject.send(sourceId)
                     }
                 }
-                self.audioTrackActivityObservationDictionary[source.sourceId] = audioTrackActivityObservation
-                tasks.append(audioTrackActivityObservation)
             }
+            self.audioTrackActivityObservationDictionary[sourceId] = audioTrackActivityObservation
+            await audioTrackActivityObservation.value
```
[the remark this answers is not in the document]
Comment (aravind-raveendran) [reply LocalPackages/RTSCore/Sources/RTSCore/Models/SourceBuilder.swift:150]: Yes @Agustin-Bologna-Dolby - we need to await the value. It's a miss that I missed awaiting in the video track activity though. Thanks for picking this up, I missed it when splitting this into two different functions at the last minute .

File: LocalPackages/RTSCore/Sources/RTSCore/Models/StreamStatistics.swift [hunk 4]
```diff
@@ -81,9 +81,9 @@ extension StreamStatistics {
 
 extension StatsInboundRtp {
     init(_ stats: MCInboundRtpStreamStats, codecStatsList: [MCCodecsStats]?) {
-        kind = stats.kind as String
-        sid = stats.sid as String
-        mid = stats.mid as String
+        kind = stats.kind as? String
+        sid = stats.sid as? String
+        mid = stats.mid as? String
```
[the remark this answers is not in the document]
Comment (aravind-raveendran) [reply LocalPackages/RTSCore/Sources/RTSCore/Models/StreamStatistics.swift:86]: Very true !! :)

File: LocalPackages/RTSCore/Sources/RTSCore/Models/SourceBuilder.swift [hunk 5]
```diff
@@ -89,24 +66,22 @@ final class SourceBuilder {
...
         videoTrackStateUpdateSubject
-            .sink { [weak self] sourceId, active in
+            .sink { [weak self] sourceId in
                 guard let self, let source = self.partialSources.first(where: { $0.sourceId == sourceId }) else {
                     return
                 }
-                Self.logger.debug("👨‍🔧 Handle video track active state change \(sourceId), \(active)")
-                source.isVideoActive = active
-                self.sources = self.makeSources()
+                Self.logger.debug("👨‍🔧 Handle video track active state change \(sourceId); isActive \(source.audioTrack?.isActive == true)")
```
[the remark this answers is not in the document]
Comment (aravind-raveendran) [reply LocalPackages/RTSCore/Sources/RTSCore/Models/SourceBuilder.swift:83]: Thank you !!

**Comment by aravind-raveendran** [comment]
> While trying the app I found the following issues:
> 
> * The streams don't have sound when they should have.

Hi @Agustin-Bologna-Dolby - Did you use a multiview stream to test this functionality ? If yes this is a correct scenario if the first source that arrives do not have an audio track associated with it ?

**Comment by aravind-raveendran** [comment]
> While trying the app I found the following issues:
> 
> * If the statistics are displayed on screen and you open the settings view the statistics panel does a weird effect.

This happened because of the stats list been reset to empty every time the view refreshes and the stats view itself was listening to the stats updates. So every time it refreshes the stats starts from empty till it update which happens every second. We can't have this view model to be a StateObject either because we have a scenario the source can go inactive and the app fallback to the next active source.

I fixed the flickering by moving the listener over to the parent view model in this case `StreamViewModel`.

**Comment by aravind-raveendran** [comment]
> > While trying the app I found the following issues:
> > 
> > * The streams don't have sound when they should have.
> 
> Hi @Agustin-Bologna-Dolby - Did you use a multiview stream to test this functionality ? If yes this is a correct scenario if the first source that arrives do not have an audio track associated with it ?

Actually yes the issue existed on single stream. Fixed !!

**Comment by aravind-raveendran** [comment]
> While trying the app I found the following issues:
> * Whenever I change the simulcast option I get redirected back to `auto`

Do you have more inputs when the simulcast layer selection redirected to `auto` ? Lets have a chat on this one ?

**Comment by aravind-raveendran** [comment]
@Agustin-Bologna-Dolby - Please take a look again when you are free !! All comments are addressed now.

**Comment by aravind-raveendran** [comment]
> > While trying the app I found the following issues:
> > 
> > * If the statistics are displayed on screen and you open the settings view the statistics panel does a weird effect.
> 
> This happened because of the stats list been reset to empty every time the view refreshes and the stats view itself was listening to the stats updates. So every time it refreshes the stats starts from empty till it update which happens every second. We can't have this view model to be a StateObject either because we have a scenario the source can go inactive and the app fallback to the next active source.
> 
> I fixed the flickering by moving the listener over to the parent view model in this case `StreamViewModel`.

While this works, it leads to massive view models when all the publishers are subscribed to in the parent view model and passed down to the Childs.
Do you have a better or cleaner idea?

**Review by Agustin-Bologna-Dolby — APPROVED** [review]

**Review by sheiladoherty-dolby — APPROVED** [review]
LGTM!

**Pull request merged**